mmdet/structures/bbox/horizontal_boxes.py

```diff
@@ -0,0 +1,400 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Optional, Tuple, TypeVar, Union
+
+import cv2
+import numpy as np
+import torch
+from torch import BoolTensor, Tensor
+
+from mmdet.structures.mask.structures import BitmapMasks, PolygonMasks
+from .base_boxes import BaseBoxes
+from .bbox_overlaps import bbox_overlaps
+from .box_type import register_box
+
+T = TypeVar('T')
+DeviceType = Union[str, torch.device]
+MaskType = Union[BitmapMasks, PolygonMasks]
+
+
+@register_box(name='hbox')
+class HorizontalBoxes(BaseBoxes):
+    """The horizontal box class used in MMDetection by default.
+
+    The ``box_dim`` of ``HorizontalBoxes`` is 4, which means the length of
+    the last dimension of the data should be 4. Two modes of box data are
+    supported in ``HorizontalBoxes``:
+
+    - 'xyxy': Each row of data indicates (x1, y1, x2, y2), which are the
+      coordinates of the left-top and right-bottom points.
+    - 'cxcywh': Each row of data indicates (x, y, w, h), where (x, y) are the
+      coordinates of the box centers and (w, h) are the width and height.
+
+    ``HorizontalBoxes`` only restores 'xyxy' mode of data. If the the data is
+    in 'cxcywh' mode, users need to input ``mode='cxcywh'`` and The code will
+    convert the 'cxcywh' data to 'xyxy' automatically.
+
+    Args:
+        data (Tensor or np.ndarray or Sequence): The box data with shape of
+            (..., 4).
+        dtype (torch.dtype, Optional): data type of boxes. Defaults to None.
+        device (str or torch.device, Optional): device of boxes.
+            Default to None.
+        clone (bool): Whether clone ``boxes`` or not. Defaults to True.
+        mode (str, Optional): the mode of boxes. If it is 'cxcywh', the
+            `data` will be converted to 'xyxy' mode. Defaults to None.
+    """
+
+    box_dim: int = 4
+
+    def __init__(self,
+                 data: Union[Tensor, np.ndarray],
+                 dtype: torch.dtype = None,
+                 device: DeviceType = None,
+                 clone: bool = True,
+                 mode: Optional[str] = None) -> None:
```

Comment:
rename it to input_mode or in_mode might be better, simply using mode may be confusing sometime. users may think it sets the box mode in the bboxe. In fact, the internal data is only xyxy